src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,374 @@
+  const [fetchSwapQuoteError, setFetchSwapQuoteError] = useState(false)
+  const [estimatedModalVisible, setEstimatedDialogVisible] = useState(false)
+  const [swapFeeModalVisible, setSwapFeeModalVisible] = useState(false)
+  const [swapInfo, setSwapInfo] = useState(null as any)
+  const coreTokens = useSelector(coreTokensSelector)
+  const walletAddress = useSelector(walletAddressSelector)
+
+  // Items set for remote config
+  const maxSlippagePercent = useSelector(maxSwapSlippagePercentageSelector)
+  const swapFeeEnabled = useSelector(swapFeeEnabledSelector)
+  const swapFeePercentage = useSelector(swapFeePercentageSelector)
+
+  // Remote configs converted to decimals strings
+  const maxSlippageDecimal = `${maxSlippagePercent / 100}`
+  const swapFeeDecimal = `${swapFeePercentage / 100}`
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  // Token Symbols
+  const toTokenSymbol = coreTokens.find((token) => token.address === toToken)?.symbol
+  const fromTokenSymbol = coreTokens.find((token) => token.address === fromToken)?.symbol
+
+  useEffect(() => {
+    if (fetchSwapQuoteError) {
+      dispatch(showError(ErrorMessages.FETCH_SWAP_QUOTE_FAILED))
+    }
+  }, [fetchSwapQuoteError])
+
+  useEffect(() => {
+    ValoraAnalytics.track(SwapEvents.swap_review_screen_open, {
+      toToken,
+      fromToken,
+      buyAmount: swapAmount[Field.TO] ?? '',
+    })
+  }, [])
+
+  const getSwapInfo = async () => {
+    try {
+      setFetchSwapQuoteError(false)
+      setLoading(true)
+      const swapAmountInWei = multiplyByWei(swapAmount[updatedField]!)
+      const swapAmountParam = updatedField === Field.FROM ? 'sellAmount' : 'buyAmount'
+      const params = {
+        buyToken: toToken,
+        sellToken: fromToken,
+        [swapAmountParam]: swapAmountInWei.toString().split('.')[0],
+        slippagePercentage: maxSlippageDecimal,
+        userAddress: walletAddress,
+      }
+      const queryParams = new URLSearchParams({ ...(params as Record<string, string>) }).toString()
+      const requestUrl = `${networkConfig.approveSwapUrl}?${queryParams}`
+      const response = await fetch(requestUrl)
+      if (!response.ok) {
+        setFetchSwapQuoteError(true)
+        Logger.error(TAG, `Failure response fetching token Swap: ${response}`)
+        throw new Error(
+          `Failure response fetching token swap quote. ${response.status}  ${response.statusText}`
+        )
+      }
+      setSwapInfo(await response.json())
+    } catch (error) {
+      setFetchSwapQuoteError(true)
+      Logger.error(TAG, `Error fetching token swap quote: ${error}`)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  // Get swap info on load
+  useAsync(getSwapInfo, [])
+
+  return (
+    <SafeAreaView style={styles.safeAreaContainer}>
+      <CustomHeader title={t('swapReviewScreen.title')} left={<BackButton />} />
+      <DisconnectBanner />
+      {(loading && swapInfo === null) || fetchSwapQuoteError ? (
```

Comment:
if for whatever reason the user sees the swap details and accidentally pulls to refresh and gets an error, it's strange that the content would be cleared...is it enough to display the error banner but keep whatever existing content still displayed?

i'd suggest doing simplifying the display to be something like 
```
<ScrollView> 
{loading && ( show loading)}
{swapInfo && (<> the swap info content </>)}
</ScrollView>
```

what do you think?